src/mc-fle2-payload-iev.c

```diff
@@ -162,9 +162,14 @@ mc_FLE2IndexedEncryptedValue_add_S_Key (_mongocrypt_crypto_t *crypto,
          mc_ServerDataEncryptionLevel1Token_get (token);
       uint32_t bytes_written;
 
-      _mongocrypt_buffer_resize (
-         &iev->Inner,
-         _mongocrypt_fle2_calculate_plaintext_len (iev->InnerEncrypted.len));
+      const uint32_t plaintext_len = _mongocrypt_fle2_calculate_plaintext_len (
+         iev->InnerEncrypted.len, status);
+      if (!plaintext_len) {
+         mc_ServerDataEncryptionLevel1Token_destroy (token);
+         CLIENT_ERR ("plaintext length must be greater than 0");
```

Comment:
```suggestion
```

Suggest removing the `CLIENT_ERR`. If `plaintext_len` is zero, `status` has been set by `_mongocrypt_fle2_calculate_plaintext_len`.